lib/Extension/LanguageServerRename/Handler/FileRenameHandler.php

```diff
@@ -39,14 +43,34 @@ public function methods(): array
     public function willRenameFiles(RenameFilesParams $params): Promise
     {
         return call(function () use ($params) {
-            $workspaceEdits = LocatedTextEditsMap::create();
-            foreach ($params->files as $rename) {
-                assert($rename instanceof FileRename);
+            $count = 0;
+            $documentChanges = [];
+            try {
+                foreach ($params->files as $rename) {
+                    $locatedEditMap = LocatedTextEditsMap::create();
 
-                $workspaceEdits = $workspaceEdits->merge(yield $this->renamer->renameFile(TextDocumentUri::fromString($rename->oldUri), TextDocumentUri::fromString($rename->newUri)));
-            }
+                    $renameEdit = yield $this->renamer->renameFile(
+                        TextDocumentUri::fromString($rename->oldUri),
+                        TextDocumentUri::fromString($rename->newUri)
+                    );
+
+                    $workspaceEdit = $this->converter->toWorkspaceEdit($renameEdit);
+
+                    foreach ($workspaceEdit->documentChanges ?? [] as $change) {
+                        $documentChanges[] = $change;
+                    }
+                }
 
-            return $this->converter->toWorkspaceEdit($workspaceEdits);
+                return new WorkspaceEdit(documentChanges: $documentChanges);
+            } catch (CouldNotRename $error) {
+                $previous = $error->getPrevious();
+
+                $this->clientApi->window()->showMessage()->error(sprintf(
+                    $error->getMessage() . ($previous?->getTraceAsString() ?? '')
```

Comment:
is `sprintf` missing a format string? `%s %s` why do we get the trace from `$previous`?

Reply:
I copied catching `CouldNotRename` from `RenameHandler` to have the same way of error handling and don't know what the trace was needed for in the original place: 

https://github.com/phpactor/phpactor/blame/a23130918b61c3f45f4a403abe56b5a06be80de9/lib/Extension/LanguageServerRename/Handler/RenameHandler.php#L93